### PR TITLE
TLS v1.3 0-RTT

### PR DESCRIPTION
**Pull request by SparkiDev**
Fixes for PSK with both session ticket and external PSK in client hello.

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -2632,9 +2700,56 @@ static int DoTls13CertificateRequest(WOLFSSL* ssl, const byte* input,
...
+    ato16(input + *inOutIdx, &len);
+    *inOutIdx += OPAQUE16_LEN;
+    if ((*inOutIdx - begin) + len > size)
+        return BUFFER_ERROR;
+    *inOutIdx += len;
+#else
 #ifdef WOLFSSL_POST_HANDSHAKE_AUTH
-    certReqCtx = XMALLOC(sizeof(CertReqCtx) + len - 1, ssl->heap,
-                         DYNAMIC_TYPE_TMP_BUFFER);
+    certReqCtx = (CertReqCtx*)XMALLOC(sizeof(CertReqCtx) + len - 1, ssl->heap,
```
Comment (dgarske) [inline src/tls13.c]: The reason for the `-1` here is not clear. Can you use OPAQUE type or add comment?

**Comment by dgarske** [comment]
Also found this warning with GCC 7.1:
```
src/tls13.c: In function 'wolfSSL_allow_post_handshake_auth':
src/tls13.c:6478:5: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
     if (ssl == NULL || !IsAtLeastTLSv1_3(ssl->version))
     ^~
src/tls13.c:6480:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
         return BAD_FUNC_ARG;
         ^~~~~~
```

**Comment by SparkiDev** [comment]
Fixed GCC 7.1 error

File: src/tls13.c (hunk 1 above)
Comment (SparkiDev) [inline src/tls13.c]: The certificate request context can be up to 255 bytes long.
wolfSSL will only use one byte but other implementations may have longer values.
The structure is over-allocated to handle this.
Comment added at allocation.

**Pull request merged**